p2p/host/basic/basic_host.go

```diff
@@ -191,12 +194,54 @@ func NewHost(ctx context.Context, net network.Network, opts *HostOpts) (*BasicHo
 		h.pings = ping.NewPingService(h)
 	}
 
+	if h.introspector == nil {
+		h.introspector = introspection.NewDefaultIntrospector()
+	}
```

Comment:
This duplicates the code in config.NewNode() [here](https://github.com/libp2p/go-libp2p/blob/b5a2ce94353a163de8f20b8e58767f711c722af3/config/config.go#L124)

We can safely assume here that an introspector exists. However, do we want to allow users to be able to disable introspection ? That decision will affect this code flow.